components/callbacks/config.go

```diff
@@ -66,3 +68,62 @@ func ConfigProvider(dc *dynamicconfig.Collection) *Config {
 		},
 	}
 }
+
+var AllowedAddresses = dynamicconfig.NewNamespaceTypedSettingWithConverter(
+	"component.callbacks.allowedAddresses",
+	allowedAddressConverter,
+	[]AddressMatchRule(nil),
+	`The per-namespace list of addresses that are allowed for callbacks and whether secure connections (https) are required.
+URLs are checked against each in order when starting a workflow with attached callbacks and only need to match one to pass validation.
+Default is no address rules, meaning all callbacks will be rejected. Any invalid entries are ignored. Each entry is a map with possible values:
+	 - "Pattern":string (required) the host:port pattern to which this config applies.
+		Wildcards, '*', are supported and can match any number of characters (e.g. '*' matches everything, 'prefix.*.domain' matches 'prefix.a.domain' as well as 'prefix.a.b.domain').
+	 - "AllowInsecure":bool (optional, default=false) indicates whether https is required`)
+
+type AddressMatchRule struct {
+	Regexp        *regexp.Regexp
+	AllowInsecure bool
+}
+
+func allowedAddressConverter(val any) ([]AddressMatchRule, error) {
+	type entry struct {
+		Pattern       string
+		AllowInsecure bool
+	}
+	intermediate, err := dynamicconfig.ConvertStructure([]entry{})(val)
```

Comment:
```suggestion
intermediate, err := dynamicconfig.ConvertStructure([]entry(nil))(val)
```

Reply:
I used `[]entry{}` instead of `[]entry(nil)` because I didn't want `AllowInsecure` to be required and have it default to false instead.

Reply:
Shouldn't those be equivalent, though? A zero-length non-nil slice and a nil slice? Does mapstructure treat them differently?